### PR TITLE
- adjust showing upcoming meetup

### LINKED ISSUE #84: Closing a picture
Click on background should also collapse an opened picture. Forcing people to looking for this small X is a rape on UX.

### PR DESCRIPTION
**Pull request by EwelinaSkrzypacz**
Before: when meetup starts, it was moved to past meetups
Now: when meetup starts, it will be moved to past meetups after midnight

Also I fixed #84.

### CODE REVIEW

File: public/assets/meetups/2024-09-19-lmt-03/data.json [hunk 1]
```diff
@@ -0,0 +1,127 @@
+{
+  "title": "Legnicki Meetup Technologiczny #3",
+  "datetime": "2024-09-19 00:15:00",
+  "place": "Legnicki Browar Książęcy",
+  "localization": "https://maps.app.goo.gl/MFfX39G6NrAerXWd6",
+  "lineup": [
+    {
+      "name": "Pxdddddd",
```
Comment (dawidrudnik) [inline public/assets/meetups/2024-09-19-lmt-03/data.json:8]: Who ? :eyes: 

```suggestion
      "name": "Przemysław",
```

Comment (EwelinaSkrzypacz) [reply public/assets/meetups/2024-09-19-lmt-03/data.json]: It was by mistake; I added the file just to test things. I deleted this file

File: public/application.js [hunk 2]
```diff
@@ -77,9 +78,17 @@ function initialize() {
...
     loadUpcomingMeetup: function () {
-      const upcomingMeetup = this.meetups.find(
-        (meetup) => new Date(meetup.datetime) > new Date()
-      )
+      const now = new Date()
+
+      const upcomingMeetup = this.meetups.find((meetup) => {
+        const meetupDate = new Date(meetup.datetime)
+
+        const endOfDay = new Date(meetupDate)
```
Comment (dawidrudnik) [inline public/application.js:86]: Why you don't use https://momentjs.com/docs/#/query/is-after/.

File: public/application.js [hunk 3]
```diff
@@ -77,9 +78,18 @@ function initialize() {
...
-        (meetup) => new Date(meetup.datetime) > new Date()
-      )
+      const now = moment()
+      const endOfDay = moment().endOf('day')
+
+      const upcomingMeetup = this.meetups.find((meetup) => {
+        const meetupDate = moment(meetup.datetime)
+
+        return (
+          now.isSameOrBefore(endOfDay, 'day') &&
```
Comment (dawidrudnik) [inline public/application.js:88]: I'm not sure about this line ``now.isSameOrBefore(endOfDay, 'day')``. When this return false ?

```suggestion
```

File: public/application.js [hunk 4]
```diff
@@ -133,9 +146,17 @@ function initialize() {
...
+      const now = moment()
+      const endOfDay = moment().endOf('day')
+
       this.pastMeetups = this.meetups
-        .filter((meetup) => new Date(meetup.datetime) < now)
+        .filter((meetup) => {
+          const meetupDate = moment(meetup.datetime)
+          return (
+            meetupDate.isBefore(now, 'day') ||
+            (meetupDate.isSame(now, 'day') && now.isAfter(endOfDay))
```
Comment (dawidrudnik) [inline public/application.js:157]: Maybe like that ?

```suggestion
            now.isSameOrBefore(meetupDate, 'day')
```

Comment (EwelinaSkrzypacz) [reply public/application.js]: After adding this, I don't see previous meetups

File: public/application.js (hunk 3 above)
Comment (EwelinaSkrzypacz) [reply public/application.js:88]: This condition is false when the current time is out of range for the day, that is, after the end of the day.

File: public/application.js (hunk 4 above)
Comment (EwelinaSkrzypacz) [reply public/application.js]: After speaking with @dawidrudnik we fixed this

File: public/application.js (hunk 3 above)
Comment (EwelinaSkrzypacz) [reply public/application.js]: After speaking with @dawidrudnik we fixed this

**Review by dawidrudnik — DISMISSED** [review]

**Review by dawidrudnik — APPROVED** [review]

**Pull request merged**